src/main/java/com/auth0/client/mgmt/UsersEntity.java

```diff
@@ -350,4 +355,213 @@ public Request deleteMultifactorProvider(String userId, String provider) {
+     * See https://auth0.com/docs/api/management/v2#!/Users/delete_permissions
+     *
+     * @param userId the user id
+     * @param permissions a list of permission objects to remove from the user
+     * @return a Request to execute
+     */
+    public Request removePermissions(String userId, List<Permission> permissions) {
+        Asserts.assertNotNull(userId, "user id");
+        Asserts.assertNotEmpty(permissions, "permissions");
+
+        Map<String, List<Permission>> body = new HashMap<>();
+        body.put("permissions", permissions);
+
+        final String url = baseUrl
+            .newBuilder()
+            .addPathSegments("api/v2/users")
+            .addPathSegments(userId)
+            .addPathSegments("permissions")
+            .build()
+            .toString();
+        VoidRequest request = new VoidRequest(this.client, url, "DELETE");
+        request.setBody(body);
+        request.addHeader("Authorization", "Bearer " + apiToken);
+        return request;
+    }
+
+    /**
+     * Assign permissions to a user. A token with update:users is needed.
+     * See https://auth0.com/docs/api/management/v2#!/Users/post_permissions
+     *
+     * @param userId the user id
+     * @param permissions a list of permission objects to assign to the user
+     * @return a Request to execute
+     */
+    public Request addPermissions(String userId, List<Permission> permissions) {
+        Asserts.assertNotNull(userId, "user id");
+        Asserts.assertNotEmpty(permissions, "permissions");
+
+        Map<String, List<Permission>> body = new HashMap<>();
+        body.put("permissions", permissions);
+
+        final String url = baseUrl
+            .newBuilder()
+            .addPathSegments("api/v2/users")
+            .addPathSegments(userId)
+            .addPathSegments("permissions")
+            .build()
+            .toString();
+        VoidRequest request = new VoidRequest(this.client, url, "POST");
+        request.setBody(body);
+        request.addHeader("Authorization", "Bearer " + apiToken);
+        return request;
+    }
+
+    /**
+     * Get the roles associated with a user.
+     * See https://auth0.com/docs/api/management/v2#!/Users/get_roles
+     *
+     * @param userId the role id
+     * @return a Request to execute
+     */
+    public Request<List<Role>> listRoles(String userId) {
+        Asserts.assertNotNull(userId, "user id");
+        String url = baseUrl
+            .newBuilder()
+            .addPathSegments("api/v2/users")
+            .addPathSegments(userId)
+            .addPathSegments("roles").build().toString();
+        CustomRequest<List<Role>> request = new CustomRequest<>(client, url, "GET", new TypeReference<List<Role>>() {
+        });
+        request.addHeader("Authorization", "Bearer " + apiToken);
+        return request;
+    }
+
+    /**
+     * Get the roles associated with a user.
+     * See https://auth0.com/docs/api/management/v2#!/Users/get_roles
```

Comment:
missing scopes (in new line) `read:users` `read:roles`. 
Also URL should be https://auth0.com/docs/api/management/v2#!/Users/get_user_roles

Reply:
This has been implemented as suggested.